### PR TITLE
[RNMobile] Synced Patterns: Ensure title is always visible

### LINKED ISSUE #37203: Synced pattern title illegible when using block-based theme with dark background
## Description
When using a block based theme that includes a dark background color, the title and help icon of a Reusable block is very difficult to read. 

## Step-by-step reproduction instructions
1. Set a site's theme to a block-based theme with a dark background, e.g. Geologist. 
2. Add a Reusable block to the block editor canvas. 

## Expected behaviour
The Reusable block title and help icon are legible. 

## Actual behaviour
The Reusable block title and help icon are illegible due to the low contrast ratio. 

## Screenshots or screen recording (optional)
Reusable block title illegible with dark background, block-based theme

## WordPress information
- WordPress version: WordPress.com
- Gutenberg version: v12.0.1
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? No, Geologist. 

## Device information
- Device: iPhone 13 mini
- Operating system: iOS 15.1.1
- WordPress app version: v18.8

### PR DESCRIPTION
**Pull request by SiobhyB**
* Gutenberg Mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/6303

## What?

Fixes https://github.com/WordPress/gutenberg/issues/37203, by ensuring the title section of Synced Patterns is alway visible within the editor.

## Why?

There are currently certain scenarios where the title section of Synced Patterns is invisible. This happens if a block-based theme with a light background is active, as the native editor displays the light background in contrast with the block's dark mode styles. The same happens if a theme has a dark background and a device's light mode if enabled.

## How?

The logic associated with the title section's styles has been updated to use the block based theme's colour scheme.

## Testing Instructions

Follow the _preparation steps_ once for each mode (dark/light) and then proceed with the _testing steps_.

### Preparation steps

* Ensure the device is in either dark mode or light mode.
* Activate a theme with support for global styles, e.g. [Byrne](https://byrnedemo.wordpress.com/).
* In your site's /wp-admin area, navigate to the site editor (Appearance → Editor) and select a style corresponding to the opposite mode (dark background for light mode, and vice versa).
* Create a [synced pattern](https://wordpress.org/documentation/article/reusable-blocks/) with a title if you haven't already, while in the browser.
* In the app, navigate to the editor for any of the site's posts or pages.

### Testing steps

* Tap the + icon to open the block inserter and select the "Synced Patterns" tab.
* Add a synced pattern to the editor.
* Select the synced pattern and verify that you're able to see its icon, title, and the help button in the top bar.

### Testing Instructions for Keyboard

N/A, solely a visual change.

### Dark mode with light background

| Before  | After |
| ------------- | ------------- |
|  |  |

### Light mode with dark background

| Before  | After |
| ------------- | ------------- |
|  |  |

### CODE REVIEW

**Comment by SiobhyB** [comment]
> **Note**
> The PHP tests are being super flaky across all PRs at the moment. I've tried re-running them a number of times. I don't believe anything in this PR itself is causing the failures.

**Review by derekblank — APPROVED** [review]
I was able to follow the testing instructions successfully, LGTM! 🚀 

I tried restarting the failing PHP tests as well -- I was able to get one previously-failing test to pass, but one is still failing. :| l checked the failures, and agree they seem flaky and unrelated to the changes in this PR. 👍

**Comment by fluiddot** [comment]
> Note The PHP tests are being super flaky across all PRs at the moment. I've tried re-running them a number of times. I don't believe anything in this PR itself is causing the failures.

@SiobhyB I've updated the PR with `trunk` to incorporate fixes to the PHP tests (https://github.com/WordPress/gutenberg/commit/fa73f8222413ca40a81dd9a75079eac7d762ad89). I hope this makes all CI jobs to pass 🤞.

File: packages/react-native-editor/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ For each user feature we should also add a importance categorization label  to i
 -->
 
 ## Unreleased
+-   [*] Social Icons: Fix visibility of inactive icons when used with block based themes in dark mode [https://github.com/WordPress/gutenberg/pull/55398]
```
Comment (fluiddot) [inline packages/react-native-editor/CHANGELOG.md]: This entry references a different fix and [PR](https://github.com/WordPress/gutenberg/pull/55398) that hasn't been merged yet. Could we update it to reference this PR instead?

**Review by fluiddot — CHANGES_REQUESTED** [review]

File: packages/block-library/src/block/edit-title.native.js [hunk 2]
```diff
@@ -6,29 +6,33 @@ import { Text, View } from 'react-native';
...
-	);
-	const titleStyle = getStylesFromColorScheme(
-		styles.title,
-		styles.titleDark
-	);
-	const infoIconStyle = getStylesFromColorScheme(
-		styles.infoIcon,
-		styles.infoIconDark
-	);
+	const globalStyles = useContext( GlobalStylesContext );
```
Comment (fluiddot) [inline packages/block-library/src/block/edit-title.native.js]: Nitpick: I noticed we already have a function to get global styles.

https://github.com/WordPress/gutenberg/blob/c605917977350b46c7852dbdf0fd722f4c1bfe24/packages/components/src/mobile/global-styles-context/index.native.js#L89-L93

**Review by fluiddot — COMMENTED** [review]
I confirmed the changes work as expected by following the Testing Instructions 🎊 . I hold on my approval until we solve [this issue](https://github.com/WordPress/gutenberg/pull/55399#discussion_r1366711704).

File: packages/block-library/src/block/edit-title.native.js (hunk 2 above)
Comment (SiobhyB) [reply packages/block-library/src/block/edit-title.native.js]: Thanks for spotting that! Updated in https://github.com/WordPress/gutenberg/pull/55399/commits/7b1e40b62c7669539c5e40f5d9f86856d530ac8b.

File: packages/react-native-editor/CHANGELOG.md (hunk 1 above)
Comment (SiobhyB) [reply packages/react-native-editor/CHANGELOG.md]: Whoops, that's me attempting to juggle too many similar changes at once. :D Updated in https://github.com/WordPress/gutenberg/pull/55399/commits/cd0c45fa988142145eb3db591a881e2532ccdbb3.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

**Pull request merged**